node/runtime/src/lib.rs

```diff
@@ -533,3 +536,38 @@ impl_runtime_apis! {
 		}
 	}
 }
+
+/// Misbehavior type which takes window length as input
+/// Each variant and its data is a seperate kind
+#[derive(Copy, Clone, Eq, Hash, PartialEq, Encode, Decode)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize, Debug))]
+pub enum Misbehavior {
```

Comment:
Could be added to `node/runtime/impls.rs` to keep the top-level file concise.

Reply:
The reason why I didn't put there is because of `//! Some configurable implementations as associated type for the substrate runtime.`